### PR TITLE
Collections should not allow FileSets as members

### LINKED ISSUE #257: adding a file_set to a collection is allowed and shouldn't be
To reproduce...
```
col = Hydra::Works::Collection.new(id: 'col-1')
fs = Hydra::Works::FileSet.new(id: 'fs-1')

col.members << fs
=> [#<Hydra::Works::FileSet id: "fs-1", head: [], tail: []>]
```

Should throw an error message.  Instead, it incorrectly adds the file set to the collection.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #257

### CODE REVIEW

**Comment by mjgiarlo** [comment]
@jcoyne @escowles @tpendragon 

I'm struggling to figure out how to have collection_behavior inject its own validator into the validators it should be inheriting from hydra-pcdm. `@type_validator` is already set, so using `||=` isn't doing the trick. The code is WIP but probably not *far* off. The two CI failures are below.

```
Failures:
  1) Hydra::Works::Collection adding file_sets to collections with ordered members raises AssociationTypeMismatch
     Failure/Error: expect { collection.ordered_members = [file_set] }.to raise_error(exception)
       expected ActiveFedora::AssociationTypeMismatch but nothing was raised
     # ./spec/hydra/works/models/collection_spec.rb:109:in `block (4 levels) in <top (required)>'
  2) Hydra::Works::Collection adding file_sets to collections with unordered members raises AssociationTypeMismatch
     Failure/Error: expect { collection.members = [file_set] }.to raise_error(exception)
       expected ActiveFedora::AssociationTypeMismatch but nothing was raised
     # ./spec/hydra/works/models/collection_spec.rb:116:in `block (4 levels) in <top (required)>'
Finished in 1 minute 3.75 seconds (files took 1.18 seconds to load)
115 examples, 2 failures, 2 pending
```

File: lib/hydra/works/models/concerns/collection_behavior.rb [hunk 1]
```diff
@@ -21,6 +21,15 @@ module CollectionBehavior
       type [Hydra::PCDM::Vocab::PCDMTerms.Collection, Vocab::WorksTerms.Collection]
     end
 
+    module ClassMethods
+      def type_validator
+        Hydra::PCDM::Validators::CompositeValidator.new(
```
Comment (mjgiarlo) [inline lib/hydra/works/models/concerns/collection_behavior.rb]: This is where it's messed up. I tried following the pattern I saw in other behaviors (using `@type_validator ||=`, but something is eluding me here).

Comment (escowles) [inline lib/hydra/works/models/concerns/collection_behavior.rb]: Can you just add the new validator to the `validators` array?  Maybe after making sure it's not nil?

**Comment by tpendragon** [comment]
Taking a look at this now.

**Comment by mjgiarlo** [comment]
Thx, @tpendragon and @escowles. Btw, I may not have much time to take this up again so if either of y'all feel close to marching this over the finish line yourselves, please do.

**Comment by tpendragon** [comment]
The fix isn't pretty - the problem is temporal coupling. The association definition uses class_name.type_validators, but that isn't changed until after the include happens.

**Comment by mjgiarlo** [comment]
Oh, gross. No wonder I was so confused.

**Comment by mjgiarlo** [comment]
@tpendragon @jcoyne @escowles Is this mergeable if I squash it, or do the hijinx in `CollectionBehavior` warrant further attention?

**Comment by mjgiarlo** [comment]
(And big thanks to @tpendragon for troubleshooting this.)

**Comment by escowles** [comment]
:+1: -- though it looks like it needs a rebase

**Comment by jcoyne** [comment]
Yeah, and you need to remove `[WIP]` from the commit messages.

**Comment by mjgiarlo** [comment]
@jcoyne @escowles Ready for enmergination.

**Comment by mjgiarlo** [comment]
Much obliged, you three.

**Pull request merged**